bindings/rust/bench/benches/handshake.rs

```diff
@@ -16,10 +14,13 @@ use criterion::{
 pub fn bench_handshake_params(c: &mut Criterion) {
     fn bench_handshake_for_library<T: TlsBenchHarness>(
         bench_group: &mut BenchmarkGroup<WallTime>,
+        name: &str,
         handshake_type: HandshakeType,
         ec_group: ECGroup,
     ) {
-        bench_group.bench_function(type_name::<T>(), |b| {
+        // generate all inputs (TlsBenchHarness structs) before benchmarking handshakes
+        // timing only includes negotiation, not config/connection initialization
```

Comment:
Good idea, but that would require a little bit of a refactor to the current benching harness, especially to separate out the config initialization and the conn initialization. I have a refactor planned that includes separating those two out, and I could possibly add a bench for that then.

Reply:
I'm pretty sure that the initialization took quite a bit of time? Taking out the initialization part was one of the first things I did, so it's been a while since I benched with it.